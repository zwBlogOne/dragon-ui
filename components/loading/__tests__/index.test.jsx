import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Loading from '../index';

describe('Loading', () => {
  it('renders hidden Loading correctly', () => {
    const wrapper = render(
      <Loading visible={false}>
        这里是内容
      </Loading>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders visible Loading correctly', () => {
    const wrapper = render(
      <Loading visible>
        这里是内容
      </Loading>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
