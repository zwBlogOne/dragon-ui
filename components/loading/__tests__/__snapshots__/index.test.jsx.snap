// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders hidden Loading correctly 1`] = `
<div
  class="ui-loading"
>
  <div
    class="ui-loading-spins"
    style="display:none"
  >
    <span
      class="ui-loading-spin ui-loading-spin-first"
    />
    <span
      class="ui-loading-spin ui-loading-spin-second"
    />
    <span
      class="ui-loading-spin ui-loading-spin-third"
    />
  </div>
  <div
    class=""
  >
    这里是内容
  </div>
</div>
`;

exports[`Loading renders visible Loading correctly 1`] = `
<div
  class="ui-loading"
>
  <div
    class="ui-loading-spins"
    style="display:block"
  >
    <span
      class="ui-loading-spin ui-loading-spin-first"
    />
    <span
      class="ui-loading-spin ui-loading-spin-second"
    />
    <span
      class="ui-loading-spin ui-loading-spin-third"
    />
  </div>
  <div
    class="ui-loading-inner"
  >
    这里是内容
  </div>
</div>
`;
